tfhe/src/boolean/casting_key/mod.rs

```diff
@@ -0,0 +1,47 @@
+use crate::boolean::engine::{BooleanEngine, WithThreadLocalEngine};
+use crate::boolean::prelude::Ciphertext;
+use crate::boolean::ClientKey;
+
+use crate::core_crypto::prelude::{keyswitch_lwe_ciphertext, LweKeyswitchKeyOwned};
+
+#[cfg(test)]
+mod test_cast;
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct CastingKey {
+    pub(crate) key_switching_key: LweKeyswitchKeyOwned<u32>,
+}
+
+impl CastingKey {
+    pub fn new(ck1: &ClientKey, ck2: &ClientKey) -> Self {
+        Self {
+            key_switching_key: BooleanEngine::with_thread_local_mut(|engine| {
+                engine.new_key_switching_key(ck1, ck2)
+            }),
+        }
+    }
+
+    pub fn cast_assign(&self, ct: &Ciphertext, ct_dest: &mut Ciphertext) {
```

Comment:
Generally our `_assign` functions takes the receiver (`&mut` argument) as the first argument

Reply:
I would name it `cast_into` then

Reply:
Ok let's do this thx